resources/scponly_user.rb

```diff
@@ -0,0 +1,110 @@
+#true To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+
+resource_name :scponly_user
+
+default_action :create
+
+property :write_dir, String, default: 'incoming'
+property :public_key, String
+property :private_key, String
+property :chroot, [true, false], default: true
+property :altroot, String
+property :binaries, Array, default: %w(/bin/chgrp /bin/chmod /bin/chown /bin/ln /bin/ls /bin/mkdir /bin/mv /bin/rm /bin/rmdir /usr/bin/scp /usr/sbin/scponlyc)
+
+action :create do
+  run_context.include_recipe 'osl-scponly::default'
+
+  if new_resource.chroot
+
+    altroot = new_resource.altroot.nil? ? '/home/chroot' : new_resource.altroot
+
+    directory "#{altroot}//home/#{new_resource.name}"do
```

Comment:
The BUILDING-JAILS documentation has some vague description of the passwd file that makes it look like this is how the home directory needs to be structured for a chroot user. On second look though it seems like this might just be how the chrooted passwd entry needs to be formatted